File: test_backend_connection.py
"""
Quick test to verify backend is accessible and endpoint exists
"""
import requests

BASE_URL = "https://urchin-app-86rjy.ondigitalocean.app"

print("Testing backend connection...")
print("=" * 60)

# Test 1: Check if backend is alive
try:
    response = requests.get(f"{BASE_URL}/", timeout=10)
    print(f"✅ Backend is alive - Status: {response.status_code}")
except Exception as e:
    print(f"❌ Backend not accessible: {e}")

# Test 2: Check if the endpoint exists
try:
    response = requests.post(
        f"{BASE_URL}/api/advanced-growth/plots/create-manual",
        data={"test": "test"},
        timeout=10
    )
    print(f"✅ Endpoint exists - Status: {response.status_code}")
    print(f"Response: {response.text[:200]}")
except Exception as e:
    print(f"❌ Endpoint error: {e}")

print("=" * 60)

File: test-backend-connection.js
/**
 * Test Backend Connection
 * Run this script to verify your React Native app can connect to the live backend
 * 
 * Usage: node test-backend-connection.js
 */

const axios = require('axios');

const BACKEND_URL = 'https://urchin-app-86rjy.ondigitalocean.app';

console.log('🧪 Testing AgroShield Backend Connection...\n');
console.log(`Backend URL: ${BACKEND_URL}\n`);

async function testEndpoint(name, endpoint) {
  try {
    console.log(`Testing ${name}...`);
    const response = await axios.get(`${BACKEND_URL}${endpoint}`, {
      timeout: 10000
    });
    console.log(`✅ ${name} - Status: ${response.status}`);
    console.log(`   Response:`, JSON.stringify(response.data).substring(0, 100));
    return true;
  } catch (error) {
    if (error.response) {
      console.log(`❌ ${name} - Status: ${error.response.status}`);
      console.log(`   Error:`, error.response.data);
    } else {
      console.log(`❌ ${name} - Error:`, error.message);
    }
    return false;
  }
}

async function runTests() {
  const tests = [
    { name: 'Farms API', endpoint: '/api/farms' },
    { name: 'Village Groups Health', endpoint: '/api/village-groups/groups/health' },
    { name: 'Upload Stats', endpoint: '/api/upload/stats' },
    { name: 'Subscription Tiers', endpoint: '/api/subscription/tiers' },
    { name: 'Drone Aggregation Bundles', endpoint: '/api/drone/marketplace/aggregation-bundles' },
  ];

  let passed = 0;
  let failed = 0;

  for (const test of tests) {
    const result = await testEndpoint(test.name, test.endpoint);
    if (result) passed++;
    else failed++;
    console.log('');
  }

  console.log('═══════════════════════════════════════');
  console.log(`Test Results: ${passed} passed, ${failed} failed`);
  console.log('═══════════════════════════════════════\n');

  if (passed === tests.length) {
    console.log('🎉 All tests passed! Your backend is ready.\n');
    console.log('Next steps:');
    console.log('1. Start your React Native app: npm start');
    console.log('2. Open on your device/emulator');
    console.log('3. Test API calls from the app\n');
  } else {
    console.log('⚠️  Some tests failed. Check your backend configuration.\n');
  }
}

runTests();
